packages/server/src/api/controllers/application.ts

```diff
@@ -470,7 +470,7 @@ const destroyApp = async (ctx: any) => {
   }
 
   /* istanbul ignore next */
-  if (!env.isTest() && !isUnpublish) {
+  if (!env.isTest() && !isUnpublish && isDevAppID(appId)) {
```

Comment:
we should add a test

```ts
describe("destroyApp", ()=> {
   it("prevents deletion of prod app without unpublish", () => {})
})
```

Reply:
The issue here is that the production (only) bucket is deleted when you delete an app through the API . So if you delete the prod app ID, the dev app is still there, but you can't access it in the builder and it's essentially bricked until you delete it as well. This just makes sure that we don't blow away the bucket when the published app is deleted. Thought it easier to just make sure we don't delete the bucket on prod appIDs.

Will add a test 👍

Reply:
Makes sense! I think we should still add the validation, even if we keep the `isDevAppID` check on the `deleteApp` call, so that we are protected against any future issues like this.

Reply:
Actually yes - it is better to just set the unpublish flag in the public API endpoint - added protection to the endpoint to prevent this happening again and make sure that every delete call to that endpoint for a production appID will fail without the unpublish flag, then added a middleware to the public API application `DELETE` endpoint to append the unpublish to it automatically.